crates/api/src/comment_report.rs

```diff
@@ -48,36 +49,45 @@ impl Perform for CreateCommentReport {
 
     check_community_ban(person_id, comment_view.community.id, context.pool()).await?;
 
-    let report_form = CommentReportForm {
-      creator_id: person_id,
-      comment_id,
-      original_comment_text: comment_view.comment.content,
-      reason: data.reason.to_owned(),
-    };
-
-    let report = blocking(context.pool(), move |conn| {
-      CommentReport::report(conn, &report_form)
-    })
-    .await?
-    .map_err(|_| ApiError::err("couldnt_create_report"))?;
-
-    let comment_report_view = blocking(context.pool(), move |conn| {
-      CommentReportView::read(conn, report.id, person_id)
-    })
-    .await??;
-
-    let res = CommentReportResponse {
-      comment_report_view,
-    };
-
-    context.chat_server().do_send(SendModRoomMessage {
-      op: UserOperation::CreateCommentReport,
-      response: res.clone(),
-      community_id: comment_view.community.id,
-      websocket_id,
-    });
+    if comment_view.community.local {
+      let report_form = CommentReportForm {
+        creator_id: person_id,
+        comment_id,
+        original_comment_text: comment_view.comment.content,
+        reason: data.reason.to_owned(),
+      };
+
+      let report = blocking(context.pool(), move |conn| {
+        CommentReport::report(conn, &report_form)
+      })
+      .await?
+      .map_err(|_| ApiError::err("couldnt_create_report"))?;
+
+      let comment_report_view = blocking(context.pool(), move |conn| {
+        CommentReportView::read(conn, report.id, person_id)
+      })
+      .await??;
+
+      context.chat_server().do_send(SendModRoomMessage {
+        op: UserOperation::CreateCommentReport,
+        response: CommentReportResponse {
+          comment_report_view,
+        },
+        community_id: comment_view.community.id,
+        websocket_id,
+      });
+    } else {
+      Report::send(
+        ObjectId::new(comment_view.comment.ap_id),
+        &local_user_view.person,
+        comment_view.community.id,
+        reason.to_string(),
+        context,
+      )
+      .await?;
+    }
 
-    Ok(res)
+    Ok(())
```

Comment:
I dont see what the return value would be useful for, as the reporter himself generally wont have permission to view reports.